tools/restatectl/src/util.rs

```diff
@@ -8,13 +8,64 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+use std::fmt::{self, Display};
+
 use tonic::transport::Channel;
 
 use restate_cli_util::CliContext;
 use restate_core::network::net_util::create_tonic_channel;
-use restate_types::net::AdvertisedAddress;
+use restate_types::{logs::metadata::ProviderConfiguration, net::AdvertisedAddress};
 
 pub fn grpc_channel(address: AdvertisedAddress) -> Channel {
     let ctx = CliContext::get();
     create_tonic_channel(address, &ctx.network)
 }
+
+pub fn write_default_provider<W: fmt::Write>(
+    w: &mut W,
+    depth: usize,
+    provider: &ProviderConfiguration,
+) -> Result<(), fmt::Error> {
```

Comment:
I'm okay to merge this of course, it'd also be nice if we use kv-styled tables.

Reply:
The table definitely looks nice. This was done originally to print cluster configuration which is kinda nested and hence I thought tree view would look better to show it (specially with diff and such). We can move to table view in a follow up imho to unify the UI across entire restatectl